lib/3scale/backend/stats/delete_job_def.rb

```diff
@@ -0,0 +1,83 @@
+module ThreeScale
+  module Backend
+    module Stats
+      class DeleteJobDef
+        ATTRIBUTES = %i[service_id applications metrics users from to context_info].freeze
+        private_constant :ATTRIBUTES
+        attr_accessor(*ATTRIBUTES)
+
+        def self.attribute_names
+          ATTRIBUTES
+        end
+
+        def initialize(params = {})
+          self.class.attribute_names.each do |key|
+            send("#{key}=", params[key]) unless params[key].nil?
+          end
+        end
+
+        def run_async
+          validate
```

Comment:
Should we wait until this point to validate the attributes?
Wouldn't it make more sense to validate on `initialize()` and on the individual setters (if they need to be exposed) ?

Reply:
:+1: